clients/mist_client.go

```diff
@@ -81,6 +90,48 @@ type MistStreamInfo struct {
 	Error  string                 `json:"error,omitempty"`
 }
 
+type MistPush struct {
+	ID           int64
+	Stream       string
+	OriginalURI  string
+	EffectiveURI string
+	Stats        *MistPushStats
+}
+
+func (p *MistPush) UnmarshalJSON(data []byte) error {
+	// this field is undocumented and shows up as null everytime.
+	var unknown json.RawMessage
+	var tmp MistPush
+	if err := UnmarshalJSONArray(data, &tmp.ID, &tmp.Stream, &tmp.OriginalURI, &tmp.EffectiveURI, &unknown, &tmp.Stats); err != nil {
+		return err
+	}
+	*p = tmp
```

Comment:
Thing looks weird, why do you assign `tmp` to Receiver? I think the normal flow would be to just return `(MistPush, error)`.

Reply:
Good question. I copied it from the mapic Mist client, but I'll revisit.

Reply:
Oh, I realized why eventually -- this pattern is to implement [the `json.Unmarshaler` interface](https://pkg.go.dev/encoding/json#Unmarshaler), so this function gets called with its custom behavior when we do a normal `json.Unmarshal`. I guess the `tmp` pattern is so that we don't make any changes to the current struct while unmarshalling? But if there's an error our struct doesn't matter anymore anyway so maybe it can be made cleaner, I'll play with it.